scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/EquationSolvingAlgorithms.java

```diff
@@ -127,25 +127,56 @@ private IteratingResult nextValueAt(
         final int maxIteration)
     throws ExceededMaxIterationException, SchedulingInterruptedException
     {
+      // the number of possible values may be less than the number of iterations, so stop after all have been visited.
+      long numTimepoints = minus(minus(max.in(Duration.MICROSECONDS), min.in(Duration.MICROSECONDS)), 1);
+      long maxIters = Long.min(maxIteration, numTimepoints);
       var cur = init;
-      int i = 0;
+      long i = 0;
       do {
         //we should not come back to previously visited values
         if (!history.alreadyVisited(cur) && cur.between(min, max)) {
           i++;
           try {
             final var value = function.valueAt(cur, history);
-            return new IteratingResult(new FunctionCoordinate<>(cur, value), i);
+            return new IteratingResult(new FunctionCoordinate<>(cur, value), ((Long)i).intValue());
           } catch (DiscontinuityException e) {
             //nothing, keep iterating
           }
         }
         cur = chooseRandomX(min, max);
-        //if min == max, another call to random will have no effect and thus we should exit
-      } while(i < maxIteration && !min.isEqualTo(max));
+        //if all timepoints have been visited, another call to random will have no effect and thus we should exit
+      } while(i < maxIters);
       throw new ExceededMaxIterationException();
     }
 
+    /**
+     * Addition of longs with safety from overflow
+     */
+    private static long plus( long rd1, long rd2 ) {
```

Comment:
can you final rd1 and rd2 ? same in the minus.

Reply:
"final" modifiers added to plus() and minus()